outputs.tf

```
@@ -78,6 +78,11 @@ output "private_subnet_arns" {
   value       = aws_subnet.private[*].arn
 }
 
+output "private_subnet_azs" {
+  description = "Map of AZs to private subnet IDs"
+  value       = { for subnet in aws_subnet.private[*] : subnet.availability_zone => subnet.id... }
```

Is this a typo, or new splat syntax I haven't seen 😅 
```suggestion
  value       = { for subnet in aws_subnet.private[*] : subnet.availability_zone => subnet.id }
```

---

The ellipsis enables grouping by key, so you end up with a map like: { us-east-1 => [subnet-1, subnet-2, etc] }

---

I'd also be more keen to just output something like the following and users can slice and dice, pick and choose how they see fit

```hcl
output "private_subnets" {
  description = "Map private subnets created and their attributes"
  value       = aws_subnet.private
```

but the name conflicts with an output we have already

---

I removed the map output and added a full resource output, and updated the title of the PR